core/domain/collection_services.py

```diff
@@ -1132,17 +1115,11 @@ def load_demo(collection_id):
     """
     delete_demo(collection_id)
 
-    if not collection_domain.Collection.is_demo_collection_id(collection_id):
-        raise Exception('Invalid demo collection id %s' % collection_id)
-
     demo_filepath = os.path.join(
         feconf.SAMPLE_COLLECTIONS_DIR,
         feconf.DEMO_COLLECTIONS[collection_id])
 
-    if demo_filepath.endswith('yaml'):
```

Comment:
This is shifted to the test file as it needs to be a test to ensure all demo filepaths end with yaml.